tests/virt_autotest/xen_guest_irqbalance.pm

```diff
@@ -0,0 +1,201 @@
+use virt_autotest::utils qw(is_kvm_host guest_is_sle wait_guest_online print_cmd_output_to_file);
+use virt_utils qw(upload_virt_logs remove_vm restore_downloaded_guests);
+
+our $vm_xml_save_dir = "/tmp/download_vm_xml";
+
+sub run_test {
+    my $self = shift;
+
+    return if is_kvm_host;
+
+    save_original_guests();
+
+    foreach my $guest (keys %virt_autotest::common::guests) {
+
+        #irqbalance test run on sle12sp3+ guest
+        next if guest_is_sle($guest, '<=12-sp2');
+
+        record_info("Test $guest");
+        prepare_guest_for_irqbalance($guest);
+
+        # with irqbalance.service is enabled, SMP affinity will be distributed among CPUs automatically even if IRQs are bound to specified CPUs in SLE15SP4
+        assert_script_run "ssh root\@$guest \"systemctl stop irqbalance.service\"" if script_run("ssh root\@$guest \"systemctl status irqbalance\"") eq 0;
+
+        my $nproc = script_output("ssh root\@$guest 'nproc'");
+
+        # Bind all NIC IRQs to cpu0 in guest
+        my @nic_irqs_id = split('\n', script_output("ssh root\@$guest \"grep vif /proc/interrupts | cut -d ':' -f1 | sed 's/^ *//'\""));
+        foreach my $irq_id (@nic_irqs_id) {
+            assert_script_run("ssh root\@$guest \"echo 1 > /proc/irq/$irq_id/smp_affinity\"");
+        }
+        my @affinities_with_binding_cpu0 = get_irq_affinities_from_guest($guest, @nic_irqs_id);
+        foreach (@affinities_with_binding_cpu0) {
+            die "Failed to bind NIC IRQs to CPU0" if $_ ne 1;
+        }
+
+        # Start the irqbalance.service again
+        assert_script_run("ssh root\@$guest \"systemctl start irqbalance\"");
+
+        # Check if the SMP affinities are distributed among CPUs
+        my @affinities_with_irqbalance = get_irq_affinities_from_guest($guest, @nic_irqs_id);
+        # for SLE15SP3 and prior, IRQs are distributed on one cpu core ramdomly.
+        # So it is correct if some of them may happen to run on cpu0
+        # but it would not be correct if all IRQs were bound to cpu0.
+        if (guest_is_sle($guest, '<=15-sp3')) {
+            my $multiply = 1;
+            $multiply *= $_ foreach (@affinities_with_irqbalance);
+            die "SMP affinities were all bound to CPU0 with irqbalance.service enabled" if $multiply == 1;
+        }
+        # for SLE15SP4 (or newer release), IRQs are specified on all CPU cores
+        # fg. affinity value is '0xf' for machine with 4 CPU cores
+        else {
+            foreach (@affinities_with_irqbalance) {
+                die "It is not correct that one of the NIC IRQs was bound to cpu0 with irqbalance enabled." if $_ eq 1;
+            }
+        }
+
+        # Get the network interrupts distribution over CPUs in guest prior to network loads
+        my @initial_total_irqs_on_cpu = get_nic_irqs_distribution_from_guest($guest, $nproc);
+
+        # Generate network load in guest
+        assert_script_run("ssh root\@$guest \"ping -f -c 500 download.opensuse.org\"");
```

Comment:
I suggest you 8.8.8.8 which is always online.

Reply:
ping '8.8.8.8' will generate network packets as well? Let me try

Reply:
I did a search, '8.8.8.8' is a free DNS server provided by Google cloud in 2009. Is it a more fit than download.opensuse.org? I need think it over and welcome anyone's suggestions.

Reply:
download.opensuse.org is meant to be always online too

Reply:
I did a search, '8.8.8.8' is a free DNS server provided by Google cloud in 2009. Is it a more fit than download.opensuse.org? I need think it over and welcome anyone's suggestions.

Reply:
You can have your own decision.